### ISSUE TITLE
composer-connector-hlfv1/hlfconnection.js getChannelPeersInOrg bug

### ISSUE THREAD

**Issue by ozt88** [hyperledger/composer#4240]
## Context

``` javascript
/// composer-connector-hlfv1/hlfconnection.js (line 1203~)
getChannelPeersInOrg(peerRoles) {
    let peers = this.client.getPeersForOrgOnChannel(this.channel.getName());
    return peers.filter((cPeer) => {
        return peerRoles.every((peerRole) => {
            return cPeer.isInRole(peerRole);
        });
    });
}
```

## Expected Behavior

return peers with role information

## Actual Behavior

return peers without role information

## Possible Fix

``` javascript
/// fabric-client/lib/Client.js getPeersForOrgOnChannel :line 1825~
if(org_peer.getName() === channel_peer.getName()) {
    found_peers[org_peer.getName()] = org_peer;//to avoid Duplicate Peers
    logger.debug('%s - adding peer to list:%s',method,org_peer.getName());
}
```
this accumulates org_peers to return, but org_peer doen't have role information.
``` javascript
if(org_peer.getName() === channel_peer.getName()) {
    found_peers[org_peer.getName()] = channel_peer; //to avoid Duplicate Peers
    logger.debug('%s - adding peer to list:%s',method,org_peer.getName());
}
```
channel_peer has role information and compatibility with org_peer.
I already made an [issue on fabric jira](https://jira.hyperledger.org/browse/FAB-11071)
but they didn't listen, you can do better than me.

## Steps to Reproduce

1. call this function with proper connection

## Existing issues

- [ ] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [ ] [GitHub Issues](https://github.com/hyperledger/composer/issues)
- [ ] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

## Context

``` javascript
/// composer-connector-hlfv1/hlfqueryhandler.js (line 38~)
constructor(connection) {
    const method = 'constructor';
    LOG.entry(method);
    this.allQueryPeers = connection.getChannelPeersInOrg([FABRIC_CONSTANTS.NetworkConfig.CHAINCODE_QUERY_ROLE]);
    this.queryPeerIndex = -1;

    this.connection = connection;
    LOG.exit(method);
}
```
HLFQueryHandler constructed with allQueryPeers from getChannelPeersInOrg.
But the peers doens't have any role information.
```javascript
//fabric-client/lib/Peer.js :line 84~
isInRole(role) {
	if(!role) {
		return true;
	} else if(typeof this._roles[role] === 'undefined') {
		return true;
	} else {
		return this._roles[role];
	}
}
```
Peer.isInrole is always true.
This HLFQueryHandler can't tell the real CHAINCODE_QUERY_ROLE peers.
So every card's connectionProfile for query doesn't work

## Your Environment

* Version used: composer-connector-hlfv1 v0.19.13
* Environment name and version (e.g. Chrome 39, node.js 5.4):
* Operating System and version (desktop or mobile):
* Link to your project:

**Comment by ozt88** [comment]
https://gerrit.hyperledger.org/r/#/c/24333/

**Comment by davidkel** [comment]
node-sdk has addressed this in 1.2, also composer 0.20.0 has changed to not use that api anymore.
